src/GeniusYield/Test/Utils.hs

```diff
@@ -139,237 +98,127 @@ fakeIron = fromFakeCoin $ FakeCoin "Iron"
 -- helpers
 -------------------------------------------------------------------------------
 
-{- | Given a test name, runs the trace for every wallet, checking there weren't
-     errors.
--}
-mkTestFor :: String -> (Wallets -> GYTxMonadClb a) -> Tasty.TestTree
-mkTestFor name action =
-    testNoErrorsTraceClb v w Clb.defaultBabbage name $ do
-      asClb pureGen (w1 wallets) $ action wallets
-  where
-    v = valueFromLovelace 1_000_000_000_000_000 <>
-        fakeGold                  1_000_000_000 <>
-        fakeIron                  1_000_000_000
-
-    w = valueFromLovelace 1_000_000_000_000 <>
-        fakeGold                  1_000_000 <>
-        fakeIron                  1_000_000
-
-    wallets :: Wallets
-    wallets = Wallets (mkSimpleWallet "w1" (Clb.intToKeyPair 1))
-                      (mkSimpleWallet "w2" (Clb.intToKeyPair 2))
-                      (mkSimpleWallet "w3" (Clb.intToKeyPair 3))
-                      (mkSimpleWallet "w4" (Clb.intToKeyPair 4))
-                      (mkSimpleWallet "w5" (Clb.intToKeyPair 5))
-                      (mkSimpleWallet "w6" (Clb.intToKeyPair 6))
-                      (mkSimpleWallet "w7" (Clb.intToKeyPair 7))
-                      (mkSimpleWallet "w8" (Clb.intToKeyPair 8))
-                      (mkSimpleWallet "w9" (Clb.intToKeyPair 9))
-
-    -- | Helper for building tests
-    testNoErrorsTraceClb :: GYValue -> GYValue -> Clb.MockConfig -> String -> Clb.Clb a -> Tasty.TestTree
-    testNoErrorsTraceClb funds walletFunds cfg msg act =
-        testCaseInfo msg
-            $ maybe (pure mockLog) assertFailure
-            $ mbErrors >>= \errors -> pure (mockLog <> "\n\nError :\n-------\n" <>  errors)
-        where
-            -- _errors since we decided to store errors in the log as well.
-            (mbErrors, mock) = Clb.runClb (act >> Clb.checkErrors) $ Clb.initClb cfg (valueToApi funds) (valueToApi walletFunds)
-            mockLog = "\nEmulator log :\n--------------\n" <> logString
-            options = defaultLayoutOptions { layoutPageWidth = AvailablePerLine 150 1.0}
-            logDoc = Clb.ppLog $ Clb.mockInfo mock
-            logString = renderString $ layoutPretty options logDoc
-
-
-    mkSimpleWallet :: WalletName -> TL.KeyPair r L.StandardCrypto -> Wallet
-    mkSimpleWallet n kp =
-        Wallet
-            { walletPaymentSigningKey = paymentSigningKeyFromLedgerKeyPair kp
-            , walletNetworkId         = GYTestnetPreprod
-            , walletName              = n
-            }
+-- | General information about the test environment to help in running polymorphic tests.
+data TestInfo = TestInfo { testGoldAsset :: !GYAssetClass, testIronAsset :: !GYAssetClass, testWallets :: !Wallets }
```

Comment:
It's really important to use the test environment specific gold/iron. So tests should not hardcode an asset class because its availability depends on the test environment.

So we pass in the assets actually available to the environment at "genesis" (not actual network genesis, but test genesis) to the tests, using this structure.